File: square/src/main/java/com/raxdenstudios/square/manager/ApplicationInterceptorManager.java
package com.raxdenstudios.square.manager;

import android.app.Application;
import android.content.res.Configuration;

import com.raxdenstudios.square.interceptor.ApplicationInterceptor;
import com.raxdenstudios.square.lifecycle.ApplicationLifecycle;

/**
 * Created by Ángel Gómez on 18/12/2016.
 */

public class ApplicationInterceptorManager extends InterceptorManager<Application, ApplicationInterceptor> implements ApplicationLifecycle {

    public ApplicationInterceptorManager(Application application) {
        super(application);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onConfigurationChanged(newConfig);
        }
    }

    @Override
    public void onCreate() {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onCreate();
        }
    }

    @Override
    public void onLowMemory() {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onLowMemory();
        }
    }

    @Override
    public void onTrimMemory(int level) {
        for (ApplicationInterceptor interceptor : interceptors) {
            interceptor.onTrimMemory(level);
        }
    }

}

File: square/src/main/java/com/raxdenstudios/square/manager/ApplicationInterceptorManager.kt
package com.raxdenstudios.square.manager

import android.app.Application
import android.content.res.Configuration

import com.raxdenstudios.square.interceptor.ApplicationInterceptor
import com.raxdenstudios.square.lifecycle.ApplicationLifecycle

/**
 * Created by Ángel Gómez on 18/12/2016.
 */

class ApplicationInterceptorManager(application: Application) : InterceptorManager<Application, ApplicationInterceptor<*>>(application), ApplicationLifecycle {

    override fun onConfigurationChanged(newConfig: Configuration?) {
        interceptors.forEach { it.onConfigurationChanged(newConfig) }
    }

    override fun onCreate() {
        interceptors.forEach { it.onCreate() }
    }

    override fun onLowMemory() {
        interceptors.forEach { it.onLowMemory() }
    }

    override fun onTrimMemory(level: Int) {
        interceptors.forEach { it.onTrimMemory(level) }
    }

}
